{"_id":"a2335ec3-165b-32d9-82ad-a616eb4de913","authorName":"Sandhya Rajesh","authorEmailEncrypted":"SYxhSbenrd5dRwz+47aaSnXxPgrv+3ZkWPmrxWMtRhL4nP1h0z/fKczk1e8YKj/gIMUzQ1fIk1FszO2ZrWNkDTKSbP5mA7r8y1HcsL+GNI8ze01kXQozrt6opNoibtO2EBYS678fp4oqlFnCe4PsXF1fSqCOlt3H/f3MvWLOz3mZJd9SnNPIp3yyh877J4kpvkO22MApY4BxHXqOai9eYMKLCXU7xC0Fq6MwaS2r5zHgfBPjk6hUpwE50WGBpqEtoMh4c6trsh6HC68l9FUMlQivBqG3WUGKgCDyv4sB+GR5dTBYxLg1eWLq/tT+MiS9xVHkEYxsfGY5CNxV+PrfmuS3lw97K4ncbGN50wAw4igdPUG8jhvemidMV8H+9uWhIKUPyXAdOMDmnTAOkMYzLN7cz1ofjmE8QvfIJJ7g3fHiU2kY77e4maIV2elf4nlPm/srWSugoE7mjeoM65usFq4tiZ2puJTkwgvXn2c8qEqyKCZQu4W/2abqrNR/jiEzhnD9MMuaFAgR4M7J4a2eKJc2w9hCF0Ugt5XKIH2WgzOyEE5lSikWYnBJzF0YGWXN1CvytWEyVkV2XspLhBnG1Y1RCQ0DwX5DHUpVuWMKMuTTi7NBDXL+fqQaZlNVrCl/KN7iPRfknoXif9hiT357vM0AhKB949Z6Sm5q6Wjgugs=","authorEmailMD5":"4b69054b1075246294f18c65becf7f89","message":"I'm trying to understand async programming and I had a question. If you had a function call TestAsyncCall() as below, would one thread be used to execute DoSomethingAsync(), and another thread to execute DoSomething()?  \r\nThen when await is encountered, it would wait for DoSomethingAsync() to complete and release that thread (while also not blocking the original thread)? Do I have this right?\r\n\r\n public async void TestAsyncCall() {  \r\n Task<string> TaskResult1 = DoSomethingAsync();  \r\n string Result2 = DoSomething();  \r\n string Result1 = await TaskResult1;   \r\n }\r\n\r\n public string DoSomething() {  \r\n return \"synch\";  \r\n }\r\n\r\n public async Task<string> DoSomethingAsync() {  \r\n await Task.Delay(10000);  \r\n return \"asynch\";  \r\n }","postId":"2012_02_async-and-await-bb1cce3d-e706-3ecf-ab89-6f7945ef7d0f","replyTo":"","date":"2017-06-25T13:48:19Z","timestamp":1498398499,"authorUserId":"disqus:sandhyarajesh"}